### ISSUE TITLE
Inherited class is using methods from abstract parent with RedisJobStore

### ISSUE THREAD

**Issue by 555Russich** [agronholm/apscheduler#767]
### Things to check first

- [X] I have checked that my issue does not already have a solution in the [FAQ](https://apscheduler.readthedocs.io/en/master/faq.html)

- [x] I have searched the existing issues and didn't find my bug already reported there

- [X] I have checked that my bug is still present in the latest release

### Version

3.10.1

### What happened?

Expectation: Inherited class is using his own methods which must be implemented from abstract parent
Reality: Inherited class is using methods from abstract parent
**Note**: I notice this problem only with `RedisJobStore`

Error in `logfile.log`:
```
INFO:root:Scheduler started
INFO:root:Added job "MyAbstractClass.initiate_all" to job store "default"
INFO:root:Removed job 140d1c85738442748e98372ee0bd1a87
INFO:apscheduler.executors.default:Running job "MyAbstractClass.initiate_all (trigger: date[2023-08-05 11:51:07 MSK], next run at: 2023-08-05 11:51:07 MSK)" (scheduled at 2023-08-05 11:51:07.788467+03:00)
ERROR:apscheduler.executors.default:Job "MyAbstractClass.initiate_all (trigger: date[2023-08-05 11:51:07 MSK], next run at: 2023-08-05 11:51:07 MSK)" raised an exception
Traceback (most recent call last):
  File "/media/russich555/hdd/Programming/Freelance/YouDo/29.2pilot/venv/lib/python3.11/site-packages/apscheduler/executors/base_py3.py", line 30, in run_coroutine_job
    retval = await job.func(*job.args, **job.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/russich555/hdd/Programming/Freelance/YouDo/29.2pilot/mre/main.py", line 22, in initiate_all
    members = cls.get_required_members()
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/russich555/hdd/Programming/Freelance/YouDo/29.2pilot/mre/main.py", line 18, in get_required_members
    raise NotImplementedError
NotImplementedError
```

Same [bug description on stackoverflow](https://stackoverflow.com/questions/76837560/inherited-class-is-using-methods-from-abstract-parent-in-apscheduler-job)

### How can we reproduce the bug?
```
from abc import ABC, abstractmethod
from datetime import datetime, timedelta
import asyncio
import logging

from apscheduler.schedulers.asyncio import AsyncIOScheduler
from apscheduler.jobstores.redis import RedisJobStore


logging.basicConfig(filename='logfile.log', level=logging.INFO)
logging.getLogger('apscheduler').setLevel(logging.DEBUG)


class MyAbstractClass(ABC):
    @classmethod
    @abstractmethod
    def get_required_members(cls):
        raise NotImplementedError

    @classmethod
    async def initiate_all(cls):
        members = cls.get_required_members()
        logging.info(f'Got {members=}')
        ...


class MyImplementation(MyAbstractClass):
    @classmethod
    def get_required_members(cls):
        return ['Alex', 'Anna']

    @classmethod
    def append_to_scheduler(cls, scheduler: AsyncIOScheduler, run_date: datetime):
        return scheduler.add_job(
            func=cls.initiate_all,
            trigger='date',
            run_date=run_date
        )


redis_job_store = RedisJobStore()
scheduler = AsyncIOScheduler(jobstores={'default': redis_job_store}, logger=logging.getLogger())
scheduler.start()

run_date = datetime.now() + timedelta(seconds=5)
MyImplementation.append_to_scheduler(scheduler=scheduler, run_date=run_date)
asyncio.get_event_loop().run_forever()
```

### Workaround I use for now: pass to `initiate_all` child class as argument
```
class MyAbstractClass(ABC):
    @classmethod
    async def initiate_all(cls, child_class):
        members = child_class.get_required_members()
        logging.info(f'Got {members=}')
        ...

class MyImplementation(MyAbstractClass):
    @classmethod
    def append_to_scheduler(cls, scheduler: AsyncIOScheduler, run_date: datetime):
        return scheduler.add_job(
            func=cls.initiate_all,
            kwargs={'child_class': cls},
            trigger='date',
            run_date=run_date
        )
```

Expected output:
```
INFO:root:Scheduler started
INFO:root:Added job "MyAbstractClass.initiate_all" to job store "default"
INFO:root:Removed job d289c9dfbad2481b99585b4de0ae6cff
INFO:apscheduler.executors.default:Running job "MyAbstractClass.initiate_all (trigger: date[2023-08-05 11:55:39 MSK], next run at: 2023-08-05 11:55:39 MSK)" (scheduled at 2023-08-05 11:55:39.280433+03:00)
INFO:root:Got members=['Alex', 'Anna']
INFO:apscheduler.executors.default:Job "MyAbstractClass.initiate_all (trigger: date[2023-08-05 11:55:39 MSK], next run at: 2023-08-05 11:55:39 MSK)" executed successfully
```

**Comment by 555Russich** [comment]
One [smart guy on stackoverflow](https://stackoverflow.com/a/76888052/15637940) patched `apscheduler.util.get_callable_name` and it solved my issue. Probably it should be implemented in `master` branch?

**Comment by agronholm** [comment]
Fixed via 71c7d95f143794224cd4e19b26303871c6dcd798.

**Comment by 555Russich** [comment]
Thanks for fix! But now I got another error, because in production code split to modules. I think it's related to this issue, let's split MRE to files for reproducing bug:

`main.py`

```
from datetime import datetime, timedelta
import asyncio
import logging

from apscheduler.schedulers.asyncio import AsyncIOScheduler
from apscheduler.jobstores.redis import RedisJobStore

from impl import MyImplementation


logging.basicConfig(filename='logfile.log', level=logging.INFO)
logging.getLogger('apscheduler').setLevel(logging.DEBUG)

redis_job_store = RedisJobStore()
scheduler = AsyncIOScheduler(jobstores={'default': redis_job_store}, logger=logging.getLogger())
scheduler.start()

run_date = datetime.now() + timedelta(seconds=5)
MyImplementation.append_to_scheduler(scheduler=scheduler, run_date=run_date)
asyncio.get_event_loop().run_forever()
```
`api.py`

```
import logging
from abc import ABC, abstractmethod


class MyAbstractClass(ABC):
    @classmethod
    @abstractmethod
    def get_required_members(cls):
        raise NotImplementedError

    @classmethod
    async def initiate_all(cls):
        members = cls.get_required_members()
        logging.info(f'Got {members=}')
        ...
```

`impl.py`

```
from datetime import datetime

from apscheduler.schedulers.asyncio import AsyncIOScheduler

from api import MyAbstractClass


class MyImplementation(MyAbstractClass):
    @classmethod
    def get_required_members(cls):
        return ['Alex', 'Anna']

    @classmethod
    def append_to_scheduler(cls, scheduler: AsyncIOScheduler, run_date: datetime):
        return scheduler.add_job(
            func=cls.initiate_all,
            trigger='date',
            run_date=run_date
        )
```

Error in `logfile.log`: 
```
INFO:root:Scheduler started
INFO:root:Added job "MyImplementation.initiate_all" to job store "default"
ERROR:apscheduler.jobstores.default:Unable to restore job "b'db5bd4658d104ce78f73400879f93394'" -- removing it
Traceback (most recent call last):
  File "/media/russich555/hdd/Programming/Freelance/YouDo/29.2pilot/venv/lib/python3.11/site-packages/apscheduler/util.py", line 289, in ref_to_obj
    obj = getattr(obj, name)
          ^^^^^^^^^^^^^^^^^^
AttributeError: module 'api' has no attribute 'MyImplementation'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/media/russich555/hdd/Programming/Freelance/YouDo/29.2pilot/venv/lib/python3.11/site-packages/apscheduler/jobstores/redis.py", line 135, in _reconstitute_jobs
    jobs.append(self._reconstitute_job(job_state))
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/russich555/hdd/Programming/Freelance/YouDo/29.2pilot/venv/lib/python3.11/site-packages/apscheduler/jobstores/redis.py", line 125, in _reconstitute_job
    job.__setstate__(job_state)
  File "/media/russich555/hdd/Programming/Freelance/YouDo/29.2pilot/venv/lib/python3.11/site-packages/apscheduler/job.py", line 273, in __setstate__
    self.func = ref_to_obj(self.func_ref)
                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/russich555/hdd/Programming/Freelance/YouDo/29.2pilot/venv/lib/python3.11/site-packages/apscheduler/util.py", line 292, in ref_to_obj
    raise LookupError('Error resolving reference %s: error looking up object' % ref)
LookupError: Error resolving reference api:MyImplementation.initiate_all: error looking up object
```

I will produce PR which helped me to solve this issue.
